unit/util/range.cpp

```diff
@@ -65,3 +65,95 @@ SCENARIO("range tests", "[core][util][range]")
     }
   }
 }
+
+class move_onlyt
+{
+public:
+  move_onlyt(move_onlyt &&) = default;
+  move_onlyt &operator=(move_onlyt &&) = default;
+  move_onlyt(const move_onlyt &) = delete;
+  move_onlyt &operator=(const move_onlyt &) = delete;
+
+  explicit move_onlyt(int value) : value{value} {};
+  int value = 0;
+};
+
+bool is_odd(const move_onlyt &move_only) {
+  return move_only.value % 2 != 0;
+}
+
+SCENARIO(
+  "Range tests, with collections of move only typed values.",
+  "[core][util][range]")
+{
+  GIVEN("A vector of move only typed values.")
+  {
+    std::vector<move_onlyt> input1;
+    for(int i = 1; i <= 10; ++i)
+      input1.emplace_back(i);
+    std::vector<move_onlyt> input2;
+    for(int i = 1; i <= 10; ++i)
+      input2.emplace_back(i);
```

Comment:
:pick: input2 is not used in all the tests, consider moving it closer to the test actually using it (maybe in a second GIVEN)